connexion/api.py

```diff
@@ -313,3 +308,16 @@ def swagger_ui_static(self, filename):
         :type filename: str
         """
         return flask.send_from_directory(str(self.swagger_path), filename)
+
+    def load_spec_from_file(self, arguments, specification):
+        arguments = arguments or {}
+
+        with specification.open(mode='rb') as swagger_yaml:
+            contents = swagger_yaml.read()
+            try:
+                swagger_template = contents.decode()
+            except UnicodeDecodeError:
+                swagger_template = contents.decode('utf-8', 'replace')
+
+            swagger_string = jinja2.Template(swagger_template).render(**arguments)
+            self.specification = yaml.safe_load(swagger_string)  # type: dict
```

Comment:
This is very confusing. I would return the loaded `dict` and then in the constructor attribute the value to the `self.specification`